soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -440,6 +441,116 @@ void Randomizer::LoadMerchantMessages(const char* spoilerFileName) {
             "%gobjet mystérieux%w pour 60 Rubis?\x1B&%gOui&Non%w",
         });
 
+    //Setup for merchant text boxes
+    //Medigoron
+    //RANDOTODO: Implement obscure/ambiguous hints
+    RandomizerGet medigoronGet = this->itemLocations[RC_GC_MEDIGORON].rgID;
+    std::vector<std::string> mgItemName;
+    if (Randomizer::GetRandoSettingValue(RSK_SHUFFLE_MERCHANTS) == 2) {//If merchant hints are clear... 
+        if (medigoronGet == RG_ICE_TRAP) {
+            medigoronGet = this->itemLocations[RC_GC_MEDIGORON].fakeRgID;
+            mgItemName = {
+                this->itemLocations[RC_GC_MEDIGORON].trickName, 
+                this->itemLocations[RC_GC_MEDIGORON].trickName,
+                this->itemLocations[RC_GC_MEDIGORON].trickName
+            };
+        } else {
+            mgItemName = EnumToSpoilerfileGetName[medigoronGet];
+        }
+    } else {
+        mgItemName = {
+            "something cool",
+            "etwas cooles",
+            "un truc cool"
+        };
+    }
+         
+    CustomMessageManager::Instance->CreateMessage(
+        Randomizer::merchantMessageTableID, TEXT_MEDIGORON,
+        {
+            TEXTBOX_TYPE_BLACK,
+            TEXTBOX_POS_BOTTOM,
+            "How about buying %r&" + mgItemName[0] + "%w for %g200 rupees%w?\x1B&%gYes&No%w",
+            "Wie wäre es mit %r&" + mgItemName[1] + "%w für %g200 Rubine?%w\x1B&%gJa!&Nein!%w",
+            "Qu'est-ce que tu dirais de l'acheter pour%g200 Rubis?%w\x1B&%gOui&Non"
+        });
+    
+    //Carpet Salesman
+    //RANDOTODO: Implement obscure/ambiguous hints
+    RandomizerGet carpetGuyGet = this->itemLocations[RC_WASTELAND_BOMBCHU_SALESMAN].rgID;
+    std::vector<std::string> cgItemName;
+    std::vector<std::string> cgBoxTwoText;
+    if (!Flags_GetRandomizerInf(RAND_INF_MERCHANTS_CARPET_SALESMAN)) {
```

Comment:
I don't think this does what you intended, this method is only run once on load, so if the player hasn't gotten the item yet the "A dangerous, running object" message will never be created.

Regardless, we shouldn't need to create our own version of the original message, we should just not intercept the original message after this flag has been set.

Reply:
The issue I'm having is that unlike Medigoron, the carpet salesman doesn't have any flags to keep track of whether you bought something, so if you read the rando textbox and buy the item, the textbox won't then be replaced with original text. Any suggestions would be helpful.

Reply:
Two options come to mind:
1) We create our own textID instead of looking to override the vanilla textID, and within the carpet salesman code set the textID to our custom one unless the item has been purchased.
2) In the condition watching for `textID == CARPET_SALESMAN` just add the flag check to that same condition, if it doesn't get handled in that method it won't replace the vanilla one.